lib/Conversion/TorchToMhlo/ViewLikeOps.cpp

```diff
@@ -316,6 +342,150 @@ bool ConvertAtenViewOp<AtenReshapeOp>::getAtenViewOpSizes(
   return getListConstructElements(adaptor.shape(), dimSizes);
 }
 
+llvm::Optional<Value> unsqueezeTensor(PatternRewriter &rewriter, Operation *op,
+                                      Value tensor,
+                                      ArrayRef<int64_t> inputUnsqzDims) {
+  // Returns a new tensor with dims of size 1 inserted at the specified
+  // position.
+  //
+  // The position indices (must be high to low dimension number of the returned
+  // tensor) are specified with unsqzDims. Indices must be in-order, and in
+  // range of tensor rank. Thus, unsqueeze a rank 1 tensor with {0, 2}, {0, 1,
+  // 3}, {0, 1, 2} are all valid dimension sets, but {0, 3}, {2} are not.
+  auto dimSizes = getDimSizesOfTensor(rewriter, op, tensor);
+  auto rank = dimSizes.size();
+  size_t newRank = rank + inputUnsqzDims.size();
+  auto unsqzDims = toPositiveDims(inputUnsqzDims, newRank);
+  for (size_t k = 0; k < unsqzDims.size(); ++k) {
```

Comment:
Unless unsqzDims change during loop iterations, don't check in comparison (https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop)